### PR TITLE
Update avatar ring color

### PR DESCRIPTION
**Pull request by laminesm**
### Platforms Impacted
- [x] iOS
- [ ] macOS

### Description of changes

The `Avatar` ring no longer uses the `Avatar`'s background color in fluent 2. The ring has been updated to now use the proper tokens: `primary` for light and `tint 30` for dark mode.

### Verification

Couldn't validate for every single shared set, but here are a few cases from the demo app.

| Before                                       | After                                      |
|----------------------------------------------|--------------------------------------------|
| before_kl_light | after_kl_light |
| before_kl_dark | after_kl_dark |
| before_ns_light | after_ns_light |
| before_ns_dark | after_ns_dark |
| before_sj_light | after_sj_light |
| before_sj_dark | after_sj_dark |
| before_th_light | after_th_light |
| before_th_dark | after_th_dark |
| before_guma_light | after_guma_light |
| before_guma_dark | after_guma_dark |

### Pull request checklist

This PR has considered:
- [x] Light and Dark appearances
- [ ] iOS supported versions (all major versions greater than or equal current target deployment version)
- [ ] VoiceOver and Keyboard Accessibility
- [ ] Internationalization and Right to Left layouts
- [ ] Different resolutions (1x, 2x, 3x)
- [ ] Size classes and window sizes (iPhone vs iPad, notched devices, multitasking, different window sizes, etc)
- [ ] iPad [Pointer interaction](https://developer.apple.com/documentation/uikit/pointer_interactions)
- [x] [SwiftUI](https://developer.apple.com/tutorials/swiftui) consumption (validation or new demo scenarios needed)
- [ ] Objective-C exposure (provide it only if needed)

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/fluentui-apple/pull/1344)

### CODE REVIEW

File: ios/FluentUI/Avatar/Avatar.swift [hunk 1]
```diff
@@ -155,7 +155,9 @@ public struct Avatar: View, TokenizedControlView {
         let ringColor = !isRingVisible ? Color.clear :
         Color(dynamicColor: state.ringColor?.dynamicColor ?? ( !shouldUseCalculatedColors ?
                                                                tokenSet[.ringDefaultColor].dynamicColor :
-                                                                backgroundColor))
+                                                                CalculatedColors.ringColor(fromPrimaryText: state.primaryText,
```
Comment (edjamesmsft) [inline ios/FluentUI/Avatar/Avatar.swift:158]: Nit: the spacing here looks like it might be off by one?

**Comment by harrieshin** [comment]
this PR i believe it should be done in main branch

**Comment by harrieshin** [comment]
```
    private struct CalculatedColors {
        static func backgroundColor(), foregroundColor()
```

and now your ringColor() i think the api should be much simplier? can we refactor a bit so that the input is just (hashCode and fluenttheme)?

File: ios/FluentUI/Avatar/Avatar.swift [hunk 2]
```diff
@@ -416,6 +418,16 @@ public struct Avatar: View, TokenizedControlView {
                                 dark: GlobalTokens.sharedColors(colorSet, .tint40))
         }
 
+        static func ringColor(fromPrimaryText primaryText: String?,
+                              secondaryText: String?,
+                              fluentTheme: FluentTheme) -> DynamicColor {
```
Comment (harrieshin) [inline ios/FluentUI/Avatar/Avatar.swift:423]: dumb question : why do we need fluentTheme?

**Comment by laminesm** [comment]
> this PR i believe it should be done in main branch

This is part of the fluent 2 changes, why should it be going to main separately?

**Comment by harrieshin** [comment]
> > this PR i believe it should be done in main branch
> 
> This is part of the fluent 2 changes, why should it be going to main separately?

afaik, the fluent avatar colors didn't change in fluent2. and what you discovered is that our avatar ring isn't accessible.

File: ios/FluentUI/Avatar/Avatar.swift (hunk 2 above)
Comment (laminesm) [reply ios/FluentUI/Avatar/Avatar.swift:423]: I assume it was needed to access `GlobalTokens` before @mischreiber's `GlobalTokens` changes. I'll go ahead and remove it.

**Comment by harrieshin** [comment]
did you need to close it? you could have just rebase and update the github pR to main?

**Pull request closed without merging**